galley/pkg/config/analysis/analyzers/namespaceconflict/namespaceselector.go

```diff
@@ -0,0 +1,237 @@
+type Analyzer struct{}
+
+var _ analysis.Analyzer = &Analyzer{}
+
+var (
+	peerauthcollection    = collections.IstioSecurityV1Beta1Peerauthentications.Name()
+	requestauthcollection = collections.IstioSecurityV1Beta1Requestauthentications.Name()
+)
+
+func (a *Analyzer) Metadata() analysis.Metadata {
+	return analysis.Metadata{
+		Name:        "namespaceconflict.Analyzer",
+		Description: "Checks conditions related to conflicting namespace level resources",
+		Inputs: collection.Names{
+			peerauthcollection,
+			requestauthcollection,
+		},
+	}
+}
+
+func (a *Analyzer) Analyze(c analysis.Context) {
+	// Analyze collections.IstioSecurityV1Beta1Peerauthentications
+	c.ForEach(peerauthcollection, func(r *resource.Instance) bool {
+		x := r.Message.(*v1beta1.PeerAuthentication)
+		if x.GetSelector() != nil {
+			// If the resource has workloads associated with it, analyze for conflicts with selector
+			a.analyzeWorkloadSelectorConflicts(r, c)
+		} else {
+			// If the resource has workloads associated with it, analyze for conflicts with namespace
+			a.analyzeNamespaceWideConflicts(r, c)
+		}
+		return true
+	})
+
+	// Analyze collections.IstioSecurityV1Beta1Requestauthentications
+	c.ForEach(requestauthcollection, func(r *resource.Instance) bool {
+		x := r.Message.(*v1beta1.RequestAuthentication)
+		if x.GetSelector() != nil {
+			// If the resource has workloads associated with it, analyze for conflicts with selector
+			a.analyzeWorkloadSelectorConflicts(r, c)
+		} else {
+			// If the resource has workloads associated with it, analyze for conflicts with namespace
+			a.analyzeNamespaceWideConflicts(r, c)
+		}
+		return true
+	})
+}
+
+func (a *Analyzer) analyzeNamespaceWideConflicts(r *resource.Instance, c analysis.Context) {
+	matches := a.findMatchingNamespace(r, c)
+	// There should be only one resource associated with a selector
+	if len(matches) != 0 {
+
+		// The namespace in which we will throw the conflict
+		xNS := r.Metadata.FullName.Namespace
+
+		// Cast the message to it's respective type to report the issue correctly
+		switch v := r.Message.(type) {
+		case *v1beta1.PeerAuthentication:
+			// Throw a Peer Authentication conflict
+			// x := r.Message.(*v1beta1.PeerAuthentication)
+			m := msg.NewNamespaceResourceConflict(r, peerauthcollection.String(),
+				string(xNS), fmt.Sprintf("(ALL) Namespace: %v", xNS),
+				matches.SortedList())
+			c.Report(collections.IstioSecurityV1Beta1Peerauthentications.Name(), m)
+			return
+		case *v1beta1.RequestAuthentication:
+			// Throw an Authorization Policy conflict
+			// x := r.Message.(*v1beta1.RequestAuthentication)
+			m := msg.NewNamespaceResourceConflict(r, requestauthcollection.String(),
+				string(xNS), fmt.Sprintf("(ALL) Namespace: %v", xNS),
+				matches.SortedList())
+			c.Report(collections.IstioSecurityV1Beta1Requestauthentications.Name(), m)
+			return
+		default:
+			fmt.Printf("I don't know about type %T!\n", v)
```

Comment:
This message can be better.

Reply:
Hi @shamsher31 I added a different default message, but still I'd be happy to get your feedback or in case you have any suggestion it would be great!